### PR TITLE
Scroll the document when scrollTop message is sent from viewer

### LINKED ISSUE #6588: Scroll the document when scrollTop message is sent from Viewer
If the viewer sends a scrollTop message to the document, there is no action taken.  We would like to scroll the document from the viewer.  Can this be made available?  jridgewell suggested chenshay for this.

cc @chenshay

### PR DESCRIPTION
**Pull request by muxin**
Fix #6588

### CODE REVIEW

File: src/service/viewer-impl.js [hunk 1]
```diff
@@ -130,6 +130,9 @@ export class Viewer {
     /** @private {!Observable<!JSONType>} */
     this.broadcastObservable_ = new Observable();
 
+    /** @private {!Observable<!JSONType>} */
+    this.scrollDocObservable_ = new Observable();
```
Comment (dvoytenko) [inline src/service/viewer-impl.js]: Does it make sense to introduce more API dependencies when we are in the process of moving them out? Let's just create a generic observable for all types of messages?

Comment (muxin) [inline src/service/viewer-impl.js]: What do you mean? Is there a PR where we are moving out the API dependencies? Do you mean changing this to the following?
```
/** @private {!Observable} */
this.scrollDocObservable_ = new Observable();
```

Comment (dvoytenko) [inline src/service/viewer-impl.js]: There'd be a map of observables: `!Object<string, !Observable<!JSONType>>`. The index is the type of message. The API will be `viewer.onMessage(eventType, handler)`. The records in the map will be created from calls to this API. Thus we simply remove a switch `if (eventType == 'x') {x.fire(payload);}` to simply `if (observables[eventType]) {observables[eventType].fire(payload)}`.

**Comment by muxin** [comment]
Rebased after refactoring observables in `viewer-impl`. @dvoytenko PTAL

**Comment by muxin** [comment]
Rebased after refactoring observables in `viewer-impl`. Tested in `viewer.html`.
@dvoytenko PTAL

**Pull request merged**